### ISSUE TITLE
Identify relevant missing nationality information

### ISSUE THREAD

**Issue by SvenLieber** [kbrbe/beltrans-data-integration#83]
Nationality information for some contributors is missing. For BELTRANS either the author, illustrator or scenarist must be of Belgian nationality. If the nationality for one of these contributors is missing, they are filtered out, however, the missing nationality might be Belgian.

Thus in order to prioritize the nationality enrichment for contributors, we should identify translations where non of these contributors is identified as Belgian **and** one of these nationalities is missing (=potentially a Belgian).

- [x] identify and enrich relevant nationalities
- [x] enrich missing nationalities via existing Wikidata/ISNI identifier

**Comment by SvenLieber** [comment]
An automated way to fill the nationality gap (not necessarily relevant ones) is by using existing identifiers to look up nationality information in third party data sources, namely in Wikidata and ISNI/BnF.

On the following figure we see that more than 40% of person authorities in our whole dataset (not just the BELTRANS nationality filtered corpus) contain an ISNI identifier but no nationality, and more than 10% a Wikidata identifier but no nationality. (data points between `2022-05-23` and `2022-06-16`, because afterwards the total number was no longer all person contributors in the database, but only contributors of the nationality filtered BELTRANS corpus).

![image](https://user-images.githubusercontent.com/3501171/179760524-161040f1-ecba-4ea2-a3ef-7eb6abf3e2be.png)

The steps we can perform are as follows:

- [ ] get missing ISNI's from Wikidata
- [ ] get missing nationality from Wikidata
- [ ] get missing nationality from ISNI/BnF (BnF as a founding member of ISNI may contain most relevant identifiers in our Belgian case)

**Comment by SvenLieber** [comment]
## Status after enriching nationality via ISNI identifiers and BnF dumps

* We could extract `550` more ISNIs via Wikidata identifier from a federated query with the Wikidata query service (unfortunately this covered translation contributors not part of the BELTRANS corpus)
* We identified `9,065` person authorities with ISNI but missing nationality
  * from which we could find `5,593` in the BnF data dump
  * from which `5,034` had nationality information

This increased the overlap of contributors (see contributor stats jupyter notebook), however, only one Belgian was additionally found resulting in one additional translation.

Enriching nationality further from the ISNI API will likely not result in (many) more Belgians since our ISNI dump from the API was created based on the criterium "nationality Belgium", and thus should already contain all known Belgians.

**Comment by SvenLieber** [comment]
There seems to be an issue in the query. There are certain works that should be identified but which are not.

The script works the following way:
* for each author, illustrator, scenarist or publishing director of a book the nationality is queried
* if the nationality is empty the word MISSING is used
* a single concatenated 'allNationalities' string is created via `CONCAT`, e.g. `BelgiumFranceNetherlands`, `MISSINGBelgium`, or `FranceMISSING`
* the query filters in a HAVING clause based on 'allNationalities': there must _not_ be Belgium and there must be MISSING. For the three showed examples the third would be selected

Because a book might not have all roles, we worked with statements such as `IF(bound(?author), ...)`.
However, exactly there lies the issue:

https://github.com/kbrbe/beltrans-data-integration/blob/f71efd7c725907983bdcd7668179c8a336958cdf/data-integration/sparql-queries/get-manifestations-with-missing-nationality.sparql#L190-L199

We check _within_  one role if the nationality label is bound, but we do not check for the CONCAT if all the components are bound. Hence the variable `?allNat` (that is queried in the group as `?allNationality` and referred like this in the HAVING clause) is probably not bound. Probably because if at least one of the roles is not bound, the whole CONCAT fails.

Thus so far we only got results when all the roles were present in a book translation.
